.github/workflows/build-test.yaml

```diff
@@ -1532,9 +1551,213 @@ jobs:
+            exit -1
+          fi
+
+      - name: remove admin-console directory
+        run: rm -rf ./admin-console
+      
+      - name: install k3s
+        uses: replicatedhq/action-k3s@main
+        with:
+          version: ${{ matrix.k8s_version }}
+          arguments: ${{ startsWith(matrix.k8s_version, 'v1.22') && '--egress-selector-mode disabled' || '' }}
+      
+      - name: download base kots version
+        run: |
+          curl -LO https://github.com/replicatedhq/kots/releases/download/$BASE_KOTS_VERSION/kots_linux_amd64.tar.gz \
+            && tar zxvf kots_linux_amd64.tar.gz \
+            && mv kots kots-$BASE_KOTS_VERSION
+      
+      - name: create namespace
+        run: |
+          set +e
+          kubectl create namespace $NAMESPACE
+      
+      - name: run kots admin-console generate-manifests using base kots version with k8s context
+        run: |
+          set +e
+          ./kots-$BASE_KOTS_VERSION admin-console generate-manifests -n $NAMESPACE --shared-password password
+      
+      - name: apply the generated manifests
+        run: |
+          set +e
+          kubectl apply -f ./admin-console -n $NAMESPACE
+
+      - name: wait for the kotsadm-minio-0 pod to be created
+        run: |
+          set +e
+          COUNTER=1
+          while [ -z "$(kubectl get pods -n $NAMESPACE | grep kotsadm-minio-0)" ]; do
+            COUNTER=$[$COUNTER +1]
+            if [ $COUNTER -gt 30 ]; then
+              echo "timed out waiting for kotsadm-minio-0 pod to be created"
+              exit -1
+            fi
+            sleep 1
+          done
+      
+      - name: wait for kotsadm-minio-0 pod to be ready
+        run: |
+          set +e
+          kubectl wait --for=condition=ready --timeout=180s pod/kotsadm-minio-0 -n $NAMESPACE
+
+      - name: wait for the kotsadm deployment to be ready
+        run: |
+          set +e
+          kubectl wait --for=condition=available --timeout=300s deployment/kotsadm -n $NAMESPACE
+      
+      - name: run kots admin-console generate-manifests using new kots version with k8s context
+        run: |
+          set +e
+          ./bin/kots admin-console generate-manifests -n $NAMESPACE \
+            --shared-password password \
+            --kotsadm-registry ttl.sh \
+            --kotsadm-namespace automated-${{ github.run_id }} \
+            --kotsadm-tag 24h
+      
+      - name: validate that ./admin-console/minio-statefulset.yaml has initContainers since a migration is needed
+        run: |
+          set +e
+          if [ -z "$(grep -E 'initContainers' ./admin-console/minio-statefulset.yaml)" ]; then
+            echo "admin-console/minio-statefulset.yaml does not have initContainers"
+            exit -1
+          fi
+      
+      - name: apply the generated manifests
+        run: |
+          set +e
+          kubectl apply -f ./admin-console -n $NAMESPACE
+
+      - name: wait for the kotsadm-minio-0 pod to be created
```

Comment:
wouldn't the minio pod already exist from the previous installation?

Reply:
yup, just need to wait for it to be ready rather than created.  will remove this one and the one below